### DISCUSSION TITLE
Raspberry PI 5 e Home Assistant

### DISCUSSION THREAD

**Discussion by chip83** [indomus/forum#2830] [Hardware & Firmware]
In una discussione su un gruppo FB su HA mi hanno detto che il Raspberry PI 5 non è "compatibile" con HA.
Io ho pensato che essendo sempre su architettura arm, non penso ci siano problemi e che tutt'alpiù poteva non essere compatibile Home Assistant OS ma che non ci fossero problemi con la versione core.
Visto che voglio prendermi un raspberry e pensavo proprio al 5 per l'esigua differenza di prezzo con il 4, mi confermate il mio pensiero o è meglio puntare sulla versione precedente?

**Comment by indomus** [comment]
Ti hanno detto una cazzata. Home Assistant OS è formalmente supportato da qualche giorno; inoltre puoi serenamente installarlo su Docker.
Ciò detto: non prendere un Raspberry 5, non ha attualmente il minimo senso. Scegli [Mini PC](https://indomus.it/componenti/mini-pc-intel-nuc-e-similari/), magari basato su Alder Lake N95 o giù di lì.

**Comment by snakuzzo** [comment]
Premesso che, salvo specifiche esigenze, non ti conviene assolutamente prendere un RPI, ma ti conviene acquistare un mini PC...
HA OS, con la release del sistema operativo 11.3, gira anche su RPI5.
Core su Docker invece puoi installarlo dove ti pare...è legato a Docker...a livello di Core non ha importanza quale sia il sistema operativo o l'architettura presente perché dipendi da Docker

**Comment by chip83** [comment]
scusate ma un miniPC viene dai 150€ in su... anzi diciamo che i modelli che sono disponibili (e non sono molti) vengono anche di più (alla fine siamo sui 200€ minimo).
Un RPI5 con case, sd e ammenicoli vari viene intono ai 110€.
Capisco le prestazioni superiori, ma alla fine stiamo parlando comunque solo di farci girare HA.
Inoltre, sul sito ufficiale mi è sembrato di vedere un sistema ARM già pronto con HA installato a 100 dollari.
Quindi, sul piano economico non conviene sempre il raspberry?

**Reply by snakuzzo** [reply]
Ripeto...dipende da quali sono le tue esigenze...non è solo ed esclusivamente una questione di prezzo.
Quanto a quella roba già pronta (es. Green, Blue, Yellow etc.)... IMHO è una roba inutile che non serve a niente...è per gente che non sa nemmeno dove trovare il pulsante di accensione di un PC

**Reply by indomus** [reply]
110 euro per un Raspberry 5 (dove?) è buttare via i soldi. Mini PC della stessa fascia di prezzo [ce ne sono](https://amzn.to/3vqedi2): questo linkato è basato su N3350, ed è strasufficiente (e preferibile a Raspberry, in quanto x86_64).
Raspberry, ripeto, ora non ha senso.

**Reply by Scialla** [reply]
concordo, un conto è avere un RPi di recupero in casa e, senza spendere soldi, si usa quel che si ha ma, andare a comprarne ora uno per quello scopo, mille volte meglio un miniPC, ma anche di classe inferiore eh.... pensa che a me gira tutto su Minix con processore Atom, roba vecchissima (ma la avevo in casa e l'ho sostituito a un Rpi3b) e va alla grande. Se non devi fare transcodifica video prenditi un miniPC a basso prezzo, come ti hanno linkato. Se devi fare transcodifica allora ci va roba più potente e devi spendere qualcosa in più

**Reply by chip83** [reply]
per trascodifica video cosa intendete esattamente? cioè perchè il video prodotto dalla cam dovrebbe essere ricodificato con qualche algoritmo?

Quello che vorrei implementare io, oltre alla normale domotica, sarebbero 1-3 cam (inizio da una) da cui vorrei accedere al flusso video da HA. Secondarimanete, se la potenza di calcolo me lo consente, vorrei anche abilitare allarmi/scenari vari al riconoscimento del movimento (non penso sia necessario il riconoscimento di persone e/o animali). Mentre su una delle 3 camere (quella esterna) mi piacerebbe implementare il riconoscimento della targa dell'auto per l'apertura del cancello.
La registrazione dei flussi video invece penso possa essere fatta sulle SD card cui ogni cam è dotata (è poi possibile accedere alle registrazioni da HA?)
Per fare questo mi basterebbe un miniPC di fascia economica? 150-200€?

Ah, inoltre, sui miniPC mi pare che nella maggior parte dei casi sia installato windows, consigliate di formattare subito e metterci su uan distro di linux per HA?
Inoltre, consigliate un miniPC con architettura x86? meglio che ARM?

Domanda aggiuntiva: su un miniPC del genere posso far girare anche Batocera per il retrogaming? O per meglio dire, posso installare sia HA che Batocera (che mi pare di capire sia un OS vero e proprio) e farli funzionare contemporanemanete? Tramite Docker?

**Reply by indomus** [reply]
non è detto che i flussi video che ti arrivano dalle CAM vadano bene così come sono. framerate, risoluzione, masking: tutte attività che potresti/dovresti fare con un NVR software come motionEye o altri.
il che torna al punto: ti serve potenza. che Raspberry Pi 5 in misura limitata ha pure, non fosse che a quel punto cominicia a scaldare come una fornace, e quindi ti serve dissipazione, maggiori consumi eccetera. e quindi torniamo al punto: ha senso, visto che un buon Mini PC x86 gli fa il culo? No. Meglio un Mini PC come quelli qui suggeriti, che ti bastano e avanzano, tanto più che una microSD ti si macina in poco tempo, se la usi per molte scritture (come suggerisci di voler fare).

meglio x86 in assoluto, ci fai girare macchine virtuali e altro, è preferibile in questo scenario ad arm.

**Reply by chip83** [reply]
Ok grazie, ho ormai capito che è meglio puntare su un mini PC. Con uno di fascia economia, 150/200€ riuscirei a gestire le cose che ho elencato nel post precedente?
Inoltre riuscirei a fare girare batocera insieme ad HA? (ved. su)

**Reply by indomus** [reply]
abbondantemente. batocera non so se c'è la versione docker oppure no, cmq certamente ci può girare

**Reply by chip83** [reply]
nel caso in cui non ci fosse la versione docker dovrei per forza installare la "distro" di batocera e poi metterci sopra quella HA core?
Peccato, avrei visto più pulito mettere un linux debian ed affiancarci HA core + Batocera "core".
Cmq,  per quanto riguarda il miniPC economico, riuscirei anche a fare il riconoscimento della targa su una delle due camere (come potenza di calcolo)?

**Reply by indomus** [reply]
dipende cosa vuoi usare per farlo. se vuoi usare tensorflow, per esempio, dubito

**Reply by chip83** [reply]
beh, presuppongo che per il riconoscimento di una targa sia richiesto un algoritmo di AI. Ho provato una cosa semplice con solo opencv e neanche mi riusciva ad indendificare correttamente i contorni. 
Non c'è già qualcosa di implementato per questo scopo? motionEye di cui sento tanto parlare non ha qualcosa che potrebbe fare al caso mio?

**Reply by indomus** [reply]
motionEye è un NVR, non c'entra col riconoscimento. ti serve roba come il già citato Tensorflow o altro.
